subsys/fb/cfb.c

```diff
@@ -445,6 +445,18 @@ int cfb_framebuffer_finalize(const struct device *dev)
 		return -ENODEV;
 	}
 
+#if defined(CONFIG_CHARACTER_FRAMEBUFFER_ROTATE)
+	uint8_t temp = 0;
+	uint16_t end = ((fb->x_res * fb->y_res) / 8U);
+
+	for (size_t i = 0; i < fb->x_res * fb->y_res / 16U; i++) {
```

Comment:
Admittedly the framebuffer is not my area of expertise, but I would suggest caching the result of "fb->x_res * fb->y_res / 16U" to prevent it from being recalculated every iteration.